src/components/MyAccount/TimedLogoutModal.tsx

```diff
@@ -0,0 +1,123 @@
+/* global document */
+import React, { useEffect, useRef, useState } from "react"
+import {
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CardHeading,
+  Spacer,
+} from "@nypl/design-system-react-components"
+
+import { buildTimeLeft, deleteCookie } from "../../utils/cookieUtils"
+import { useLogoutRedirect } from "../../server/auth"
+import { useRouter } from "next/router"
+import styles from "../../../styles/components/TimedLogoutModal.module.scss"
+
+/**
+ * This renders a modal interface based on an early version from the
+ * Reservoir Design System through the `old-ds-modal` CSS class.
+ */
+const TimedLogoutModal = ({
+  stayLoggedIn,
+  expirationTime,
+  timeoutWindow = 5,
+}: {
+  expirationTime: string
+  stayLoggedIn: () => void
+  timeoutWindow?: number
+}) => {
+  const [open, setOpen] = useState(false)
+  const router = useRouter()
+  const redirectUri = useLogoutRedirect()
+  const logOutAndRedirect = () => {
```

Comment:
Ah, I thought this function lived outside of the file. So my previous suggestion of mocking this function doesn't apply.

Reply:
Yea it's a little tricky because it needs the redirect uri. I could pull it out and make the uri a param. I may try for that.

Reply:
@EdwinGuzman do you want me to try and pull this out and try to get tests running here?

Reply:
Hmm i think it might be useful to test the side effects from calling this function, but I think you can mock the external router and deleteCookie function instead and verify that they're called. Can you try that first and if that doesn't work then you can try pulling this function out and running a check that this function was called rather than the two inside of it.

Reply:
Yes, I'd be okay with that.